pandas/core/missing.py

```diff
@@ -447,7 +475,14 @@ def _akima_interpolate(xi, yi, x, der=0, axis=0):
     return P(x, nu=der)
 
 
-def _cubicspline_interpolate(xi, yi, x, axis=0, bc_type="not-a-knot", extrapolate=None):
+def _cubicspline_interpolate(
+    xi: ArrayLike,
+    yi: ArrayLike,
+    x: Union[ArrayLike, Scalar],
+    axis: Optional[int] = 0,
```

Comment:
How is axis optional? (it has a default of 0 already, so it's always specified?)

Reply:
AFAIK `Optional` means it could be `None`.

That said, the docstring lists this arg as optional so I'll fix that while I'm here

Reply:
`Optional[T]` in type-hinting is not the same as optional in docstrings. The former is an alias for `Union[T, None]` while the latter means doesn't need to be specified.